svc_test.go

```diff
@@ -131,4 +134,31 @@ func TestAutoNATServiceDialRateLimiter(t *testing.T) {
 	AutoNATServiceResetInterval = save2
 	AutoNATServiceThrottle = save3
 	manet.Private4 = save4
+	AutoNATServiceResetJitter = save5
+}
+
+func TestAddrToIP(t *testing.T) {
```

Comment:
Could you add test for this branch in the function ?

```
if strings.HasPrefix(n, "tcp") || strings.HasPrefix(n, "udp") {
		ip, _, err = net.SplitHostPort(ip)
```

Reply:
the first two test cases split host and port. It looks like `manet.DialArgs()` fails to handle degenerate multiaddrs with a protocol but not port, like `/ip4/172.16.0.1/udp`.

I'm not convinced we end up with the `SplitHostPort` actually erroring after the current constraints of `DialArgs` if that's the branch you want tested